craft_providers/multipass/multipass.py

```diff
@@ -0,0 +1,382 @@
+            ) from error
+
+        data_list = json.loads(proc.stdout).get("list", dict())
+        return [instance["name"] for instance in data_list]
+
+    def mount(
+        self,
+        *,
+        source: pathlib.Path,
+        target: str,
+        uid_map: Dict[str, str] = None,
+        gid_map: Dict[str, str] = None,
+    ) -> None:
+        """Mount host source path to target.
+
+        :param source: Path of local directory to mount.
+        :param target: Target mount points, in <name>[:<path>] format, where
+            <name> is an instance name, and optional <path> is the mount point.
+            If omitted, the mount point will be the same as the source's
+            absolute path.
+        :param uid_map: A mapping of user IDs for use in the mount of the form
+            <host-id> -> <instance-name-id>.  File and folder ownership will be
+            mapped from <host> to <instance-name> inside the instance_name.
+        :param gid_map: A mapping of group IDs for use in the mount of the form
+            <host-id> -> <instance-name-id>.  File and folder ownership will be
+            mapped from <host> to <instance-name> inside the instance_name.
+        """
+        command = ["mount", str(source), target]
+
+        if uid_map is not None:
+            for host_id, instance_id in uid_map.items():
+                command.extend(["--uid-map", f"{host_id}:{instance_id}"])
+
+        if gid_map is not None:
+            for host_id, instance_id in gid_map.items():
+                command.extend(["--gid-map", f"{host_id}:{instance_id}"])
+
+        try:
+            self._run(command, capture_output=True, check=True)
+        except subprocess.CalledProcessError as error:
+            raise MultipassError(
+                brief=f"Failed to mount {str(source)!r} to {target!r}.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+    def start(self, *, instance_name: str) -> None:
+        """Start VM instance.
+
+        :param instance_name: the name of the instance to start.
+
+        :raises MultipassError: on error.
+        """
+        command = ["start", instance_name]
+
+        try:
+            self._run(command, check=True)
+        except subprocess.CalledProcessError as error:
+            raise MultipassError(
+                brief=f"Failed to start VM {instance_name!r}.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+    def stop(self, *, instance_name: str, delay_mins: int = 0) -> None:
+        """Stop VM instance.
+
+        :param instance_name: the name of the instance_name to stop.
+        :param delay_mins: Delay shutdown for specified number of minutes.
+
+        :raises MultipassError: on error.
+        """
+        command = ["stop"]
+
+        if delay_mins != 0:
+            command.extend(["--time", str(delay_mins)])
+
+        command.append(instance_name)
```

Comment:
Why not include the instance name in the `command` definition above (as other methods do)?

Reply:
Fair question, I fixed the ordering :D